### PR TITLE
feat(dataframe): design structure and interfaces of Pandas extensions

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,3 +3,4 @@
 #
 
 tensorbay >= 1.20.0
+pandas
```
Comment (linjiX) [inline requirements.txt:6]: the pandas minimal version is needed.

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/pandas_extension.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+from pandas.api.extensions import (
+    ExtensionArray,
+    ExtensionDtype,
+    ExtensionScalarOpsMixin,
+    register_extension_dtype,
+)
+
+
+@register_extension_dtype
+class GravitiDtype(ExtensionDtype):  # type: ignore[misc]
```
Comment (linjiX) [inline graviti/dataframe/pandas_extension.py:22]: what mypy error is ignored here?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/pandas_extension.py [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        pass
+
+    @classmethod
+    def _from_sequence(  # pylint: disable=arguments-differ
+        cls, scalars: Sequence[Any], *, dtype: Optional[GravitiDtype] = None, copy: bool = False
+    ) -> "GravitiSeries":
+        """Construct a new GravitiSeries from a sequence of scalars.
+
+        Arguments:
+            scalars : Each element will be an instance of the scalar type for this
```
Comment (graczhual) [inline graviti/dataframe/pandas_extension.py:175]: do not need space here. same error elsewhere.

**Review by graczhual — CHANGES_REQUESTED** [review]

**Comment by AChenQ** [comment]
Don't use pandas extensions.

**Pull request closed without merging**